### ISSUE TITLE
onMessageReceived function not called on iOS

### ISSUE THREAD

**Issue by tolpit** [NativeScript/push-plugin#32]
Hi,

on iOS the callback in onMessageReceived isn't called when the app is in the background or close. 
Is it the expected behavior ? In my case, the tap on the notification should open a new url inside a webview, and it's only launch the app.

**Comment by AntonDobrev** [comment]
@tolpit 

You are correct, the onMessageReceived callback is not called when the app is not in the foreground. It will be called once the user taps on the notification - you can read more here - http://docs.telerik.com/platform/backend-services/javascript/push-notifications/push-handle#event-availability

**Comment by tolpit** [comment]
Ho yes, I don't expect to call the onMessageReceived callback when the app is in the foreground.

When the app is in the foreground, and the user taps on the notifcation, it open the app, and then it should fire the onMessageReceived callback, but it isn't at this moment neither.

**Comment by slavaGanzin** [comment]
> When the app is in the foreground, and the user taps on the notifcation, it open the app, and then it should fire the onMessageReceived callback, but it isn't at this moment neither.
Same on Android 4.4

**Comment by tolpit** [comment]
Some details about how I try to register the push notifications :

```
var pushsettings = {
    // Android settings
    senderID: '00000000000', // Android: Required setting with the sender/project number
    notificationCallbackAndroid: notificationHandle,

    // iOS settings
    badge: true,
    sound: false,
    alert: true,

    notificationCallbackIOS: notificationHandle
};

pushPlugin.register(pushsettings,
    // Success callback
    function(token) {
        if(pushPlugin.onMessageReceived) {
            pushPlugin.onMessageReceived(pushsettings.notificationCallbackAndroid);
        }
        
        console.log('Device registered successfully');
    },
    // Error Callback
    function(error){
        console.trace(error);
    }
);

//The callback who isn't called when the app is closed or in the background :/
function notificationHandle(message, data) {
    if(application.android) {
        ...
    }
    else if(application.ios) {
        //Save the user come from notification, to open webview with this url
        applicationSettings.setBoolean("fromNotification", true);
        applicationSettings.setString("notificationURL", message.urlActu); //urlActu is the url to show inside the webview
        
        //Should set or change the webview url
        pageData.set("url",  message.urlActu);
    }  
}
```

**Comment by AntonDobrev** [comment]
@tolpit @slavaGanzin 

This behavior is not expected. Please note that if your app is cold-started from a push notification you need to call _register_ again to supply the JS callbacks to the native code. This way you will be able to receive the notification event in your callback. 

Can you please confirm that you are receiving notifications in the notification tray? 

@tolpit 

I am not sure if the code in the _notificationHandle_ function will run correctly - are you able to call it when the app is opened and you send a push to it? 

Are you able to correctly register for push notifications - obtain a token and save it to your server?

Can you supply a simple project that illustrates the issue?

**Comment by tolpit** [comment]
The register is correctly called at every start of the app and the notifications are receive correctly when the app is in the foreground.

Yes the _notificationHandle_ run as expect when the app is in the foreground and a push notification is send to the application. 

Do you want an access to the repo ? (private gitlab)

**Comment by AntonDobrev** [comment]
@tolpit 

Thanks for specifying this. I managed to reproduce the issue locally (at the moment for iOS only). With the other collaborators we will need some time to investigate this and will get back to you.

Let me know should you find further relevant information.

**Comment by slavaGanzin** [comment]
@AntonDobrev Push notifications appears in system tray. But i can't get theirs content. Can you explain:
> to call register again to supply the JS callbacks to the native code

What is register?

**Comment by AntonDobrev** [comment]
@slavaGanzin 

I meant that your app needs to call the _register_ method of the plugin each time the app is starting or at a suitable place at the beginning of the app (for example, when the user signs in), so that the JS callbacks are dispatched to the app and can receive the native event arguments. You can also update the device registration on the server at this point. 

For example:

```
application.on(application.launchEvent, function (args) {
	
	var pushSettings = {
		....
	};

//	register for push here
});
```

However, I was able to reproduce the behavior (see my previous reply) and will come back with more details in the beginning of the next week.

**Comment by slavaGanzin** [comment]
@AntonDobrev Thanks for clarification

**Comment by tolpit** [comment]
Hello back, any update on this issue ? Did you reproduce the same problem ?

**Comment by srcmax** [comment]
Hello,
Hope to have a bug fix on this issue in the coming days?
Thanks !

**Comment by ilyago** [comment]
@tolpit Did your resolve how to invoke tap action to load url in webview when app is in the background?

**Comment by AntonDobrev** [comment]
Yes, the problem was reproduced and will be addressed. I cannot commit to a time-frame for now, but hopefully this will be delivered soon.

**Comment by AntonDobrev** [comment]
Fixed in [0.0.16](https://github.com/NativeScript/push-plugin/tree/0.0.16)
